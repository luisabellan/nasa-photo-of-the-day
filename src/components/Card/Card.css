/* colors */
html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
  border: 0;
  font: inherit;
  font-size: 14px;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

/* HTML5 display-role reset for older browsers */
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}

body {
  line-height: 1.3;
  margin: 0;
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

.container {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  background-color: #0c0c01;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  height: 100%;
}

.container .hidden {
  display: none;
}

.container .show {
  cursor: pointer;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.container .up {
  margin-bottom: 5rem;
}

.container .up h1 {
  color: #a16306;
  font-size: 3.6rem;
  font-weight: bold;
  margin-top: 5rem;
}

.container .middle-container {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.container .middle-container .left {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -ms-flex-item-align: center;
      align-self: center;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  padding: 0 10rem;
  width: 40%;
}

.container .middle-container .left .date-container {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

.container .middle-container .left .date-container .date {
  -ms-flex-item-align: end;
      align-self: flex-end;
  color: #a16306;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3rem;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  margin-right: 5px;
}

.container .middle-container .left h3 {
  color: #a16306;
  font-weight: bold;
  margin-bottom: 1rem;
}

.container .middle-container .left p {
  background-color: #0c0c01;
  color: #a16306;
  font-family: "Leckerli One", cursive;
  font-size: 1.2rem;
  width: 100%;
}

.container .middle-container .right {
  -ms-flex-item-align: start;
      align-self: flex-start;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  padding: 0 20rem;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -ms-flex-item-align: center;
      align-self: center;
}

.container .middle-container .right img {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-item-align: center;
      align-self: center;
  height: 20rem;
  margin: 0;
  width: 100%;
  max-width: 40rem;
  margin-top: 3rem;
}

.container .middle-container .right p {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-item-align: center;
      align-self: center;
  color: #a16306;
  font-size: 2rem;
  margin-top: 3rem;
}

.container .bottom {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  margin: 5rem 20rem;
  height: 100%;
}

.container .bottom button.date-picker {
  background-color: #a16306;
  border-radius: 5px;
  color: #0c0c01;
  font-size: 1.2rem;
  height: 5rem;
  margin-top: 15px;
}

.container .bottom button.date-picker span {
  color: #0c0c01;
  font-weight: bold;
  cursor: pointer;
  font-size: 2rem;
  margin-right: 2px;
  height: 4rem;
}

.container .bottom form {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  margin-top: 3rem;
}

.container .bottom form input {
  color: #a16306;
  cursor: pointer;
  font-size: 23px;
  text-align: center;
}
/*# sourceMappingURL=Card.css.map */