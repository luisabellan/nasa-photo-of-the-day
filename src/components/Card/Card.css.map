{
    "version": 3,
    "mappings": "AAAA,YAAY;AASZ,AAAA,IAAI;AACJ,IAAI;AACJ,GAAG;AACH,IAAI;AACJ,MAAM;AACN,MAAM;AACN,MAAM;AACN,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,CAAC;AACD,UAAU;AACV,GAAG;AACH,CAAC;AACD,IAAI;AACJ,OAAO;AACP,OAAO;AACP,GAAG;AACH,IAAI;AACJ,IAAI;AACJ,GAAG;AACH,GAAG;AACH,EAAE;AACF,GAAG;AACH,GAAG;AACH,GAAG;AACH,CAAC;AACD,CAAC;AACD,IAAI;AACJ,KAAK;AACL,MAAM;AACN,MAAM;AACN,GAAG;AACH,GAAG;AACH,EAAE;AACF,GAAG;AACH,CAAC;AACD,CAAC;AACD,CAAC;AACD,MAAM;AACN,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,EAAE;AACF,QAAQ;AACR,IAAI;AACJ,KAAK;AACL,MAAM;AACN,KAAK;AACL,OAAO;AACP,KAAK;AACL,KAAK;AACL,KAAK;AACL,EAAE;AACF,EAAE;AACF,EAAE;AACF,OAAO;AACP,KAAK;AACL,MAAM;AACN,OAAO;AACP,KAAK;AACL,MAAM;AACN,UAAU;AACV,MAAM;AACN,MAAM;AACN,MAAM;AACN,IAAI;AACJ,GAAG;AACH,MAAM;AACN,IAAI;AACJ,OAAO;AACP,OAAO;AACP,IAAI;AACJ,IAAI;AACJ,KAAK;AACL,KAAK,CAAC;EACJ,MAAM,EAAE,CAAC;EACT,IAAI,EAAE,OAAO;EACb,SAAS,EAAE,IAAI;EACf,MAAM,EAAE,CAAC;EACT,OAAO,EAAE,CAAC;EACV,cAAc,EAAE,QAAQ;CACzB;;AACD,iDAAiD;AACjD,AAAA,OAAO;AACP,KAAK;AACL,OAAO;AACP,UAAU;AACV,MAAM;AACN,MAAM;AACN,MAAM;AACN,MAAM;AACN,IAAI;AACJ,GAAG;AACH,OAAO,CAAC;EACN,OAAO,EAAE,KAAK;CACf;;AAED,AAAA,IAAI,CAAC;EACH,WAAW,EAAE,GAAG;EAChB,MAAM,EAAE,CAAC;CACV;;AAED,AAAA,EAAE;AACF,EAAE,CAAC;EACD,UAAU,EAAE,IAAI;CACjB;;AAED,AAAA,UAAU;AACV,CAAC,CAAC;EACA,MAAM,EAAE,IAAI;CACb;;AAED,AAAA,UAAU,AAAA,OAAO;AACjB,UAAU,AAAA,MAAM;AAChB,CAAC,AAAA,OAAO;AACR,CAAC,AAAA,MAAM,CAAC;EACN,OAAO,EAAE,IAAI;CACd;;AAED,AAAA,KAAK,CAAC;EACJ,eAAe,EAAE,QAAQ;EACzB,cAAc,EAAE,CAAC;CAClB;;AAMD,AAAA,UAAU,CAAC;EACT,WAAW,EAAE,MAAM;EACnB,gBAAgB,EA/IV,OAAc;EAgJpB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EAGtB,eAAe,EAAE,MAAM;EACvB,MAAM,EAAE,IAAI;CAwIb;;AAhJD,AAUE,UAVQ,CAUR,OAAO,CAAC;EACN,OAAO,EAAE,IAAI;CACd;;AAZH,AAcE,UAdQ,CAcR,KAAK,CAAC;EACJ,MAAM,EAAE,OAAO;EACf,OAAO,EAAE,IAAI;CACd;;AAjBH,AAoBE,UApBQ,CAoBR,GAAG,CAAC;EACF,aAAa,EAAE,IAAI;CAQpB;;AA7BH,AAuBI,UAvBM,CAoBR,GAAG,CAGD,EAAE,CAAC;EACD,KAAK,EAtKG,OAAe;EAuKvB,SAAS,EAAE,MAAM;EACjB,WAAW,EAAE,IAAI;EACjB,UAAU,EAAE,IAAI;CACjB;;AA5BL,AA+BE,UA/BQ,CA+BR,iBAAiB,CAAC;EAChB,WAAW,EAAE,MAAM;EACnB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,MAAM;CAgExB;;AAnGH,AAqCI,UArCM,CA+BR,iBAAiB,CAMf,KAAK,CAAC;EACJ,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,UAAU,EAAE,MAAM;EAClB,IAAI,EAAE,CAAC;EACP,OAAO,EAAE,OAAO;EAChB,KAAK,EAAE,GAAG;CA2BX;;AAtEL,AA6CM,UA7CI,CA+BR,iBAAiB,CAMf,KAAK,CAQH,eAAe,CAAC;EACd,WAAW,EAAE,MAAM;EACnB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;CAUvB;;AA1DP,AAkDQ,UAlDE,CA+BR,iBAAiB,CAMf,KAAK,CAQH,eAAe,CAKb,KAAK,CAAC;EACJ,UAAU,EAAE,QAAQ;EACpB,KAAK,EAlMD,OAAe;EAmMnB,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,MAAM;EACjB,eAAe,EAAE,UAAU;EAC3B,YAAY,EAAE,GAAG;CAClB;;AAzDT,AA4DM,UA5DI,CA+BR,iBAAiB,CAMf,KAAK,CAuBH,EAAE,CAAC;EACD,KAAK,EA3MC,OAAe;CA4MtB;;AA9DP,AA+DM,UA/DI,CA+BR,iBAAiB,CAMf,KAAK,CA0BH,CAAC,CAAC;EACA,gBAAgB,EA7MhB,OAAc;EA8Md,KAAK,EA/MC,OAAe;EAgNrB,WAAW,EAAE,uBAAuB;EACpC,SAAS,EAAE,MAAM;EACjB,KAAK,EAAE,IAAI;CACZ;;AArEP,AAwEI,UAxEM,CA+BR,iBAAiB,CAyCf,MAAM,CAAC;EACL,UAAU,EAAE,UAAU;EACtB,OAAO,EAAE,IAAI;EACb,IAAI,EAAE,CAAC;EACP,OAAO,EAAE,OAAO;EAChB,cAAc,EAAE,MAAM;EACtB,UAAU,EAAE,MAAM;CAoBnB;;AAlGL,AAgFM,UAhFI,CA+BR,iBAAiB,CAyCf,MAAM,CAQJ,GAAG,CAAC;EACF,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,MAAM;EAClB,MAAM,EAAE,KAAK;EACb,MAAM,EAAE,CAAC;EACT,KAAK,EAAE,IAAI;EACX,SAAS,EAAE,KAAK;EAChB,UAAU,EAAE,IAAI;CAEjB;;AAzFP,AA2FM,UA3FI,CA+BR,iBAAiB,CAyCf,MAAM,CAmBJ,CAAC,CAAC;EACA,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,MAAM;EAClB,KAAK,EA5OC,OAAe;EA6OrB,SAAS,EAAE,IAAI;EACf,UAAU,EAAE,IAAI;CACjB;;AAjGP,AAqGE,UArGQ,CAqGR,OAAO,CAAC;EACN,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,MAAM;EACvB,WAAW,EAAE,MAAM;EACnB,MAAM,EAAE,UAAU;EAClB,MAAM,EAAE,IAAI;CAoCb;;AA/IH,AA6GI,UA7GM,CAqGR,OAAO,CAQL,MAAM,AAAA,YAAY,CAAC;EACjB,gBAAgB,EA5PR,OAAe;EA6PvB,aAAa,EAAE,GAAG;EAClB,KAAK,EA7PH,OAAc;EA8PhB,SAAS,EAAE,MAAM;EACjB,MAAM,EAAE,IAAI;EACZ,UAAU,EAAE,IAAI;CAWjB;;AA9HL,AAqHM,UArHI,CAqGR,OAAO,CAQL,MAAM,AAAA,YAAY,CAQhB,IAAI,CAAC;EACH,KAAK,EAnQL,OAAc;EAoQd,WAAW,EAAE,IAAI;EACjB,MAAM,EAAE,OAAO;EACf,SAAS,EAAE,IAAI;EACf,YAAY,EAAE,GAAG;EACjB,MAAM,EAAE,IAAI;CAEb;;AA7HP,AAgII,UAhIM,CAqGR,OAAO,CA2BL,IAAI,CAAC;EACH,WAAW,EAAE,MAAM;EACnB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,MAAM;EACvB,UAAU,EAAE,IAAI;CASjB;;AA9IL,AAuIM,UAvII,CAqGR,OAAO,CA2BL,IAAI,CAOF,KAAK,CAAC;EACJ,KAAK,EAtRC,OAAe;EAuRrB,MAAM,EAAE,OAAO;EACf,SAAS,EAAC,IAAI;EACd,UAAU,EAAE,MAAM;CAEnB",
    "sources": [
        "Card.scss"
    ],
    "names": [],
    "file": "Card.css"
}