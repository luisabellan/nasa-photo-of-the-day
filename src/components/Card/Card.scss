/* colors */
$dark-orange: rgb(161, 99, 6);
$black: rgb(12, 12, 1);
// styles

// reset:
// http://meyerweb.com/eric/tools/css/reset/    v2.0 | 20110126
// License: none (public domain)

html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
  border: 0;
  font: inherit;
  font-size: 14px;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}

body {
  line-height: 1.3;
  margin: 0;
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}





.container {
  align-items: center;
  background-color: $black;
  display: flex;
  flex-direction: column;
 

  justify-content: center;
  height: 100%;

  .hidden {
    display: none;
  }

  .show {
    cursor: pointer;
    display: flex;
  }

  
  .up {
    margin-bottom: 5rem;

    h1 {
      color: $dark-orange;
      font-size: 3.6rem;
      font-weight: bold;
      margin-top: 5rem;
    }
  }

  .middle-container {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;

    .left {
      display: flex;
      flex-direction: column;
      align-self: center;
      flex: 1;
      padding: 0 10rem;
      width: 40%;

      .date-container {
        align-items: center;
        display: flex;
        flex-direction: column;

        .date {
          align-self: flex-end;
          color: $dark-orange;
          display: flex;
          font-size: 1.3rem;
          justify-content: flex-start;
          margin-right: 5px;
        }
      }

      h3 {
        color: $dark-orange;
        font-weight: bold;
        margin-bottom: 1rem;
      }
      p {
        background-color: $black;
        color: $dark-orange;
        font-family: "Leckerli One", cursive;
        font-size: 1.2rem;
        width: 100%;
      }
    }

    .right {
      align-self: flex-start;
      display: flex;
      flex: 1;
      padding: 0 20rem;
      flex-direction: column;
      align-self: center;
     
      img {
        display: flex;
        align-self: center;
        height: 20rem;
        margin: 0;
        width: 100%;
        max-width: 40rem;
        min-width: 30rem;
        margin-top: 3rem;
        
      }

      p {
        display: flex;
        align-self: center;
        color: $dark-orange;
        font-size: 2rem;
        margin-top: 3rem;
      }
    }
  }

  .bottom {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 5rem 20rem;
    height: 100%;

    button.date-picker {
      background-color: $dark-orange;
      border-radius: 5px;
      color: $black;
      font-size: 1.2rem;
      height: 5rem;
      margin-top: 15px;

      span {
        color: $black;
        font-weight: bold;
        cursor: pointer;
        font-size: 2rem;
        margin-right: 2px;
        height: 4rem;
        
      }
    }

    form {
      align-items: center;
      display: flex;
      flex-direction: column;
      justify-content: center;
      margin-top: 3rem;

      input {
        color: $dark-orange;
        cursor: pointer;
        font-size:23px;
        text-align: center;
      
      }
    }
  }
}
