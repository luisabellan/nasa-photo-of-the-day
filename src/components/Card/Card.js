import {useState, useEffect } from "react"
import axios from "axios"
import "./Card.scss"

import Moment from "react-moment"
import moment from "moment"

import DatePicker from "react-datepicker"
import "react-datepicker/dist/react-datepicker.css"
// this comment tells babel to convert jsx to calls to a function called jsx instead of React.createElement
/** @jsx jsx */
import { jsx } from '@emotion/core'
// this comment tells babel to convert jsx to calls to a function called jsx instead of React.createElement
/** @jsx jsx */
import styled from '@emotion/styled'




function Card() {
  const [data, setData] = useState({})
  const [startDate, setStartDate] = useState(new Date())

  
  useEffect(() => {
    axios
      .get(
        `https://api.nasa.gov/planetary/apod?api_key=KeyfXJ9CeZYid6vFLabJkgH3vMNJqYea6veDdo1j`
      )
      .then(res => setData(res.data))
      .catch(err => `Houston we have an error: ${err}`)      
  }, []);

  const toggleDate = e => {
    e.preventDefault();
    const input = document.getElementById("toggle")
    if (input.className.match("hidden")) {
      input.className = "show"
    } else {
      input.className = "hidden"
    }
  };

  const onChange = date => {
    setStartDate(date)
    //console.log(startDate)
    let fancyDate = moment(date).format("YYYY-MM-DD")
    //console.log(fancyDate)
    axios
      .get(
        `https://api.nasa.gov/planetary/apod?api_key=KeyfXJ9CeZYid6vFLabJkgH3vMNJqYea6veDdo1j&date=${fancyDate}`
      )
      .then(res => setData(res.data))
      .catch(err => console.log(err))
  };

  const TopDiv = styled.div`

      



  `
  const MiddleDiv = styled.div`


  `
  const BottomDiv = styled.div`

  `

  const Title = styled.h1`
  font-family: 'Merienda', cursive;
  border: 3px solid  rgb(161, 99, 6);


  `

  const DateDiv = styled.div`
  font-family: 'Poppins', sans-serif;
  
  `
  const DescriptionH3 = styled.h3`
  font-family: 'Leckerli One', cursive;
  `
  const ExplanationP = styled.p`
  font-family: 'Merienda', cursive;
  border-right: 3px solid  rgb(161, 99, 6);
  border-left: 3px solid  rgb(161, 99, 6);
  padding: 0 2rem;
  `
  
  const PhotoImg = styled.img`
  border: 2px solid rgb(161, 99, 6);
  border-radius: 12px;
  `
  /* const AuthorP = styled.p` ` */
 
  const DatePickerButton = styled.button`
  
  `


//console.log(data);

  return (
    <div className="container">
      <TopDiv className="up">
        <Title>{data.title}</Title>
      </TopDiv>

      <MiddleDiv className="middle-container">
        <div className="first">
          <PhotoImg alt={data.title} src={data.hdurl} />
          <DateDiv className="date-container">
            <Moment className="date" format="MMMM Do YYYY">
              {data.date}
            </Moment>
          </DateDiv>
          <DescriptionH3>Description</DescriptionH3>
          <ExplanationP> {data.explanation}</ExplanationP>
        </div>

        <div className="second">
          {/* <AuthorP>Author: {data.copyright}</AuthorP> */}
        </div>
      </MiddleDiv>

      <BottomDiv className="bottom date-picker">
        <DatePickerButton className="date-picker" onClick={toggleDate}>
          <span>Choose date</span>
        </DatePickerButton>

        <form>
          <DatePicker
            selected={startDate}
            id="toggle"
            className="hidden"
            onChange={onChange}
          />
        </form>
      </BottomDiv>
      <footer>
        <p>
          &copy; Copyright - Luis Abellan {moment({ startDate }).format("YYYY")}
        </p>
      </footer>
    </div>
  );
}

export default Card
