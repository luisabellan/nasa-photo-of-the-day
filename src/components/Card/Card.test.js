import React from "react";
import Card from "../Card/Card";
import renderer from "react-test-renderer";

test("Card changes date of article", () => {
  const card = renderer.create(
    <Card />
  );
  let tree = card.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onClick();

  // re-rendering
  tree = card.toJSON();
  expect(tree).toMatchSnapshot();
});
/* 
The test fails because there is an expected change between snapshots: the date for the articles changes because even though it's finally shown in a this format: January 13th 2020, it is taken from the API in UTC format, and that includes hours, minutes and seconds, and those change from snapshot to snapshot.
As no unexpected changes occur we can consider the test valid and update the snapshot by pressing 'u' after yarn test produces de output. (The date syntax is managed by moment.js).

According to the docs at https://jestjs.io/docs/en/snapshot-testing :

Snapshot Testing

Snapshot tests are a very useful tool whenever you want to make sure your UI does not change unexpectedly.

A typical snapshot test case for a mobile app renders a UI component, takes a snapshot, then compares it to a reference snapshot file stored alongside the test. The test will fail if the two snapshots do not match: either the change is unexpected, or the reference snapshot needs to be updated to the new version of the UI component.
 */