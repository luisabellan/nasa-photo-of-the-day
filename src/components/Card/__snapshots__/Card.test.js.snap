// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card changes date of article 1`] = `
<div
  className="container"
>
  <div
    className="up css-q2fyr7"
  >
    <h1
      className="css-75xtvi"
    />
  </div>
  <div
    className="middle-container css-1g37tb9"
  >
    <div
      className="first"
    >
      <div
        className="date-container css-41j3rb"
      >
        <time
          className="date"
          dateTime={"2020-01-14T01:22:39.924Z"}
        >
          January 14th 2020
        </time>
      </div>
      <h3
        className="css-14ln1wo"
      >
        Description
      </h3>
      <p
        className="css-sfzhzy"
      >
         
      </p>
    </div>
    <div
      className="second"
    >
      <img
        className="css-17muvj7"
      />
      <p
        className="css-175m5k6"
      >
        Author: 
      </p>
    </div>
  </div>
  <div
    className="bottom css-10jth8l"
  >
    <button
      className="date-picker css-175m5k6"
      onClick={[Function]}
    >
      <span>
        Choose date
      </span>
    </button>
    <form>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="hidden"
            disabled={false}
            id="toggle"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value="01/14/2020"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
